src/lib/Microsoft.Health.Fhir.Ingest.Template/CalculatedFunction/JmesPathExpressionEvaluator.cs

```diff
@@ -0,0 +1,66 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using DevLab.JmesPath;
+using EnsureThat;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.Ingest.Template.CalculatedFunction
+{
+    public class JmesPathExpressionEvaluator : IExpressionEvaluator
+    {
+        private JmesPath.Expression _jmespathExpression;
+
+        public JmesPathExpressionEvaluator(
+            JmesPath.Expression jmespathExpression)
+        {
+            _jmespathExpression = EnsureArg.IsNotNull(jmespathExpression, nameof(jmespathExpression));
+        }
+
+        public JToken SelectToken(JToken data)
+        {
+            EnsureArg.IsNotNull(data);
+            var jmesPathArgument = _jmespathExpression.Transform(data);
```

Comment:
Does this throw and if so should this be wrapped in a try catch with a specific throw exception?

Reply:
This doesn't throw. If no data can be found it will return a JToken with a type of 'Null'.